### PR TITLE
refactor: use timeout type when possible

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

ref: #5404 

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/04-channel/keeper/timeout.go [hunk 1]
```diff
@@ -68,10 +68,9 @@ func (k Keeper) TimeoutPacket(
 		return err
 	}
 
-	timeoutHeight := packet.GetTimeoutHeight()
-	if (timeoutHeight.IsZero() || proofHeight.LT(timeoutHeight)) &&
-		(packet.GetTimeoutTimestamp() == 0 || proofTimestamp < packet.GetTimeoutTimestamp()) {
-		return errorsmod.Wrap(types.ErrPacketTimeout, "packet timeout has not been reached for height or timestamp")
```
Comment (DimitrisJim) [inline modules/core/04-channel/keeper/timeout.go:74]: tangential: `types.ErrPacketTimeout` is solely used further down in `timeout.go` atm (see https://github.com/cosmos/ibc-go/blob/6c6ce879fc0201af907b864f955ce509222e12ab/modules/core/04-channel/keeper/timeout.go#L181).

Considering we've changed timeouts here and in `packet.go` to refer to `ErrTimeoutElapsed`, does it make sense to replace that one too (we now have unified timeout handling for both packets + channel upgrades)

**Review by DimitrisJim — APPROVED** [review]
lgtm, love the clean up! Left one question w.r.t `ErrPacketTimeout`.

File: modules/core/04-channel/keeper/timeout.go (hunk 1 above)
Comment (colin-axner) [reply modules/core/04-channel/keeper/timeout.go:74]: good question. I'm gonna leave as is. I don't have access to the proof height/timestamp in TimeoutExecuted and given that this is the error string in the upgrade, it likely should say packet timeout. I could do something like `errorsmod.Wrap(types.ErrTimeoutElapsed, "packet timed out")`. That or leaving as `types.ErrPacketTimeout` is fine by me

Comment (colin-axner) [reply modules/core/04-channel/keeper/timeout.go:74]: on second thought, I added the wrapped error

Comment (DimitrisJim) [reply modules/core/04-channel/keeper/timeout.go:74]: beautiful!

**Review by charleenfei — APPROVED** [review]
nice cleanup! thanks @colin-axner

**Review by damiannolan — APPROVED** [review]
Nice and clean! Thanks @colin-axner 🚀

**Comment by damiannolan** [comment]
Added backport label, I'm assuming we want it!

File: modules/core/04-channel/keeper/timeout.go (hunk 1 above)
Comment (DimitrisJim) [reply modules/core/04-channel/keeper/timeout.go:74]: I'll open issue re: deprecation later if no one beats me to it else this easily ends up forgotten

**Pull request merged**